discrete_optimization/pickup_vrp/solver/lp_solver_pymip.py

```diff
@@ -2010,45 +2008,6 @@ def update_model_lazy(
                         >= 1
                     )
                 ]
-    if len_component_global > 1 and False:
```

Comment:
Here : the piece of code should still exist. the False has been added to improve performance for some experiments and I should have removed it. We should pass an argument to the function, by default False and use it in this condition.

Reply:
So by default, this code block must be ignored? (with the argument at False)
Or the opposite (since you say that you should have removed the `and False`) ?
What could be the name of the variable holding this value ? `thorough_update` ? So that if True, we pass potentially in this block and if False not?

Reply:
I tried something (see the 2 last commits). I put by default True because you said that you should have removed the "and False". But you said also that the argument should be False by default so i am not so sure about it...

Reply:
I don't see when "thorough_update" is used in the core of the function ? 
`len_component_global = len(components_global)
    list_constraints = []
    if len_component_global > 1:
        print("Nb component : ", len_component_global)
        for s in components_global:
            # print(s)
            edge_in_of_interest = [e for n in s[0]
                                   for e in edges_in_all_vehicles[n]
                                   if e[1][0] not in s[0]
                                   and e[1][1] in s[0]]
            edge_out_of_interest = [e for n in s[0]
                                    for e in edges_out_all_vehicles[n]
                                    if e[1][0] in s[0]
                                    and e[1][1] not in s[0]]
            if not any(problem.target_vehicle[v] in s[0] for v in problem.origin_vehicle):
                list_constraints += [lp_solver.model.add_lazy_constr(mip.quicksum([lp_solver.variable_decisions["variables_edges"][e[0]][e[1]]
                                                                                   for e in edge_out_of_interest]) >= 1)]
            if not any(problem.origin_vehicle[v] in s[0] for v in problem.origin_vehicle):
                list_constraints += [lp_solver.model.add_lazy_constr(mip.quicksum([lp_solver.variable_decisions["variables_edges"][e[0]][e[1]]
                                                                                   for e in edge_in_of_interest]) >= 1)]
    if len_component_global > 1 and thorough_update:
        constraints_order = {}
        try:
            variable_order = lp_solver.variable_order
`

Reply:
and by default, thorough_update wouls be false, so the behaviour would be the same as the original code. We can rename this parameter with "do_constraint_on_order", i'll have to documentate it later.

Reply:
Done